content/docs/rfcs/75/README.md

```diff
@@ -0,0 +1,207 @@
+Existing methods present in Waku Store protocol are based on full Waku message delivery using time range query where there is no way to know apart from a given timestamp range if other Waku messages are also missing, making Store only method unreliable and bandwidth consuming.
+Operator that wants to synchronize sends a request to its peer, and in return peer operator send the response.
+After a brief low-bandwidth consuming back and flow exchange, resulting will be a list message hashes that the requesting operator needs to sync with the peer operator.
+These messages hashes can be used by other parts of the Store protocol to make sure the related full messages are wired to the requesting operator and stored in its accessible (local for now) Store/DB.
+Making the requesting operator fully synchronized with its peer operator.
+
+# Theory / Semantics
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+Consider a request-response protocol with two roles: a client and a server.
+Client and Server both MUST be peers supporting the Waku Sync protocol.
+Each operator MUST contain empty or populated Prolly tree of Waku message hashes.
+There is no other eligibility for the node at this point in time to be a client or a server who can synchronize with each other.
+
+# Wire Format Specification / Syntax
+
+```
+    syntax = "proto3";
+
+    package prollytree;
+
+    message Node {
+        bytes data = 1;              // The data stored in the node
+        int64 timestamp = 2;          // Timestamp or other unique identifier
+        bytes node_hash = 3;         // Hash of the node
+        int32 level = 4;              // Level of the node in the tree
+        bytes merkle_hash = 5;       // Merkle hash of the node
+        bool is_tail = 6;             // Flag indicating if it's a tail node
+
+        // Relationships (IDs or indices of related nodes)
+        // Note: Relationships like left, right, up, and down are typically represented using references
+        // (like indices or IDs) rather than directly embedding the objects to avoid deep nesting.
+        // In our case we can use the key of the node itself and figure out the level based on the current node level.
+        int64 left = 7;               // Reference to the left node
+        int64 right = 8;              // Reference to the right node
+        int64 up = 9;                 // Reference to the parent node
+        int64 down = 10;              // Reference to the child node
+    }
+
+    // Request message for GetRootAtHeight
+    message ExchangeRootAtHeightRequest {
+        Node root = 1;    // The root node of the Prolly tree
+        int32 height = 1;  // The desired height
```

Comment:
Do we see any reason for this `height` to be different to the `level` inside the `Node` field? If not, perhaps the protocol can stipulate that the desired height is equal to the `level` of the root node in the request?

Reply:
Good suggestion! let me make this change.